engine/table/src/main/java/io/deephaven/engine/table/impl/locations/TableDataService.java

```diff
@@ -20,6 +20,22 @@ public interface TableDataService {
     @NotNull
     TableLocationProvider getTableLocationProvider(@NotNull TableKey tableKey);
 
+    /**
+     * Request a raw {@link TableLocationProvider} from this service. A raw {@link TableLocationProvider} does not
+     * compose and/or delegate to other {@link TableLocationProvider}s.
+     *
+     * @param tableKey The {@link TableKey} to lookup
+     * @param tableLocationKey The {@link TableLocationKey} to lookup
+     * @return A raw {@link TableLocationProvider} for the specified {@link TableKey} and {@link TableLocationKey}, or
+     *         null if either key is not present
+     * @implSpec Non-raw {@link TableLocationProvider}s must implement this method.
```

Comment:
i don't understand this implspec - TableLocationProvider doesn't have this method, TableDataService does.

Reply:
Meant to state "TableDataService". Fixed; thx for catching.